### PR TITLE
Improve test coverage on datasets functions

### LINKED ISSUE #49: Add programmatic releases
The command make dist could also create a release in Github

https://developer.github.com/v3/repos/releases/#create-a-release

### PR DESCRIPTION
**Pull request by ivyleavedtoadflax**
## Description

Following #149 this PR adds additional tests for the dataset downloader functions. NB, other tasks in #149 will be completed in separate PRs.

- [x] Improve datasets/conll.py coverage to > 90%

## Checklist

- [x] Added link to Github issue or Trello card
- [x] Added tests

### CODE REVIEW

**Comment by ivyleavedtoadflax** [comment]
@aCampello @lizgzil @nsorros  Can we cut turn this into four seperate PRs? I can't guarantee I will get around to all four, but this one at least is ready to merge https://github.com/wellcometrust/WellcomeML/issues/149

File: wellcomeml/datasets/conll.py [hunk 1]
```diff
@@ -48,29 +53,35 @@ def _load_data_spacy(data_path, inc_outside=True):
...
+            "test_conll".
+    """
+    path = check_cache_and_download(dataset)
+
+    map = {"train": "eng.train", "test": "eng.testa", "evaluate": "eng.testb"}
+
+    try:
+        data_path = os.path.join(path, map[split])
+        X, Y = _load_data_spacy(data_path, inc_outside=inc_outside)
+    except KeyError:
```
Comment (aCampello) [inline wellcomeml/datasets/conll.py:83]: This is the only lines being missed from the tests are these ones. I think you could very quickly add

```python
with pytest.raises(KeyError):
    load_conll(split='wrong_argument')
````

if you want to raise your coverage even more

**Review by aCampello — APPROVED** [review]
Looks good as far as I am concerned. There is an option to raise coverage even more by adding a `pytest.raises` test, if you want, in the comments.

File: wellcomeml/datasets/conll.py (hunk 1 above)
Comment (ivyleavedtoadflax) [reply wellcomeml/datasets/conll.py:83]: ooh nice :+1:

Comment (ivyleavedtoadflax) [reply wellcomeml/datasets/conll.py:83]: Resolved in https://github.com/wellcometrust/WellcomeML/pull/154/commits/180fedf6e63d651b9ca482209114d678fe3d1547

Comment (aCampello) [reply wellcomeml/datasets/conll.py:83]: Cool, but now you got a flake8 pipeline problem :). After you fix that, and the tests pass, feel free to merge.

**Review by aCampello — CHANGES_REQUESTED** [review]
The tests now failed. I think you need to `import pytest` at the beginning of the test.

**Comment by ivyleavedtoadflax** [comment]
Ah that's weird, i had added it locally but not pushed :facepalm:

**Review by aCampello — APPROVED** [review]
:shipit:

**Pull request merged**